src/OrchardCore.Modules/OrchardCore.Forms/Views/Items/ReCaptchaPart.Fields.Edit.cshtml

```diff
@@ -0,0 +1,14 @@
+@using OrchardCore.Forms.Drivers
+@using OrchardCore.Forms.ViewModels
+@model ReCaptchaPartEditViewModel
+
+@{
+    if (Model.SettingsAreConfigured)
+    {
+        <div class="alert alert-success"><a class="alert-link" target="_blank" asp-controller="Admin" asp-action="Index" asp-area="OrchardCore.Settings" asp-route-groupId="@ReCaptchaSettingsDisplay.GroupId">reCaptcha settings</a> are configured.</div>
```

Comment:
Localization

Reply:
Yeah, I wanted to ask about this. As you can see, I'm including HTML as part of the alert text. When I localize that using `T`, the HTML gets encoded since it's of type `IHtmlLocalizer`. Should I inject an `IStringLocalizer` instead?

Reply:
Because I believe that if I localize `"reCaptcha settings"` separately from `"are configured"`, it's not going to look well for RTL.

Reply:
So if the HTML is to be part of the localized string, then that string should be as clean as possible. E.g. it should not contain a `class`, `target` and `href` attribute. So this is what I'm thinking I should do:

```csharp
@using Microsoft.Extensions.Localization
@using OrchardCore.Forms.Drivers
@using OrchardCore.Forms.ViewModels
@model ReCaptchaPartEditViewModel
@inject IStringLocalizer S
@{
    var settingsUrl = Url.Action("Index", "Admin", new { area = "OrchardCore.Settings", groupId = ReCaptchaSettingsDisplay.GroupId });
    var successMessageTemplate = "<a {0}>reCaptcha settings</a> are configured.";
    var warningMessageTemplate = "<a {0}>reCaptcha settings</a> are not configured. The captcha will not be displayed.";
    var hyperLinkAttributes = $"class=\"alert-link\" target=\"_blank\" href=\"{settingsUrl}\"";
    var successMessage = S[successMessageTemplate, hyperLinkAttributes];
    var warningMessage = S[warningMessageTemplate, hyperLinkAttributes];
}
@{
    if (Model.SettingsAreConfigured)
    {
        <div class="alert alert-success">@successMessage</div>
    }
    else
    {
        <div class="alert alert-warning">@warningMessage</div>
    }
}
```

Reply:
Never mind - I actually should use T, not S.

Reply:
So it will be:

```aspnet
@using Microsoft.Extensions.Localization
@using OrchardCore.Forms.Drivers
@using OrchardCore.Forms.ViewModels
@model ReCaptchaPartEditViewModel
@{
    var settingsUrl = Url.Action("Index", "Admin", new { area = "OrchardCore.Settings", groupId = ReCaptchaSettingsDisplay.GroupId });
    var successMessageTemplate = "<a {0}>reCaptcha settings</a> are configured.";
    var warningMessageTemplate = "<a {0}>reCaptcha settings</a> are not configured. The captcha will not be displayed.";
    var hyperLinkAttributes = $"class=\"alert-link\" target=\"_blank\" href=\"{settingsUrl}\"";
    var successMessage = T[successMessageTemplate, hyperLinkAttributes];
    var warningMessage = T[warningMessageTemplate, hyperLinkAttributes];
}
@{
    if (Model.SettingsAreConfigured)
    {
        <div class="alert alert-success">@successMessage</div>
    }
    else
    {
        <div class="alert alert-warning">@warningMessage</div>
    }
}
```